weekly-scan/template.yaml

```diff
@@ -0,0 +1,191 @@
+apiVersion: "v1"
+kind: "Template"
+metadata:
+  name: "weekly-scan-job-template"
+objects:
+  - apiVersion: "v1"
+    kind: "BuildConfig"
+    metadata:
+      name: ${PIPELINE_NAME}
+    spec:
+      source:
+        type: Git
+        git:
+          uri: ${GIT_URL}
+          ref: ${GIT_BRANCH}
+      strategy:
+        type: "JenkinsPipeline"
+        jenkinsPipelineStrategy:
+          jenkinsfile: |
+            properties([
+                pipelineTriggers([
+                    cron('0 23 * * 6'),
+                ])
+            ])
+            podTemplate(
+                cloud: 'openshift',
+                name: 'ccp-pipeline',
+                label: 'ccp-pipeline',
+                serviceAccount: 'jenkins',
+                containers: [
+                  containerTemplate(
+                    name: 'jnlp',
+                    image: 'registry.centos.org/pipeline-images/ccp-openshift-slave',
+                    ttyEnabled: true,
+                    alwaysPullImage: true,
+                    workingDir: '/tmp',
+                    privileged: true,
+                    args: '${computer.jnlpmac} ${computer.name}'
+                  )
+                ],
+                volumes: [
+                  hostPathVolume(
+                    hostPath: '/var/run/docker.sock',
+                    mountPath: '/var/run/docker.sock'
+                  )
+                ]
+            )
+            {
+                node('ccp-pipeline') {
+                    def image_name = "${APP_ID}/${JOB_ID}:${DESIRED_TAG}"
+                    def image_name_with_registry = "${REGISTRY_URL}/${image_name}"
+                    def registry_tags = "http://${REGISTRY_URL}/v2/${APP_ID}/${JOB_ID}/tags/list"
+                    def daemonset_name = "scan-data_scan-data"
+                    def image_in_registry = false
+                    def image_pulled = false
+                    def success = false
+                    try{
+                        stage('Check image exists in registry'){
+                            sh "curl ${registry_tags} | grep ${DESIRED_TAG}"
+                            image_in_registry = true
+                        }
+                        stage('Pull the docker image'){
+                            if (image_in_registry==true) {
+                                sh "docker pull ${image_name_with_registry}"
+                                image_pulled = true
+                            }
+                            else {
+                                sh "echo 'Not pulling image as its not in registry' && exit 1"
+                            }
+                        }
+                        stage('Scan the image') {
+                            parallel (
+                                "RPM updates": {
+                                  if (image_pulled==true) {
```

Comment:
this is already being checked in earlier stage, we don't need to add another check here as stages are running in sequence and it would exit earlier if image is absent.

Reply:
Once again I think this is not required. If image pull has failed, we break the flow and skip to sending mail. No point continuing further and having numerous `echo` statements!